x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -110,30 +118,43 @@ export const createMetricThresholdExecutor = (libs: InfraBackendLibs) =>
       executionId,
     });
 
-    const {
-      alertWithLifecycle,
-      savedObjectsClient,
-      getAlertUuid,
-      getAlertStartedDate,
-      getAlertByAlertUuid,
-    } = services;
+    const { alertsClient, savedObjectsClient } = services;
+    if (!alertsClient) {
+      throw new Error(`Expected alertsClient to be defined but it was not!`);
+    }
 
-    const alertFactory: MetricThresholdAlertFactory = (
+    const alertReporter: MetricThresholdAlertReporter = async (
       id,
       reason,
       actionGroup,
+      contextWithoutAlertDetailsUrl,
       additionalContext,
       evaluationValues
-    ) =>
-      alertWithLifecycle({
+    ) => {
+      const { uuid, start } = alertsClient.report({
         id,
-        fields: {
+        actionGroup,
+      });
+
+      alertsClient.setAlertData({
+        id,
+        payload: {
           [ALERT_REASON]: reason,
-          [ALERT_ACTION_GROUP]: actionGroup,
```

Comment:
action group is set by the framework so does not need to be explicitly reported back